### PR TITLE
Use OpenTelemetry attribute keys from upstream

### PR DESCRIPTION
**Pull request by tcnghia**
# Description

Update the OpenTelemetry attribute keys by using the "go.opentelemetry.io/otel/semconv" package.  This will update the *outdated* attribute keys `db.{type,url,instance}` to the latest spec `db.{system,connection_string,name}`. Other keys are kept the same.

This SDK is not GA, so we don't want to depend on the library widely (yet), but we are depending on their spec already so the use is justified here. Plus we localize the usage to a single package (tracing).

Since these attributes already changed before without us knowing, also add a unit test here that hopefully will give us some idea when the next change (if any - hopefully, none) happens.

## Issue reference

Please reference the issue this PR will close: #2288 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/diagnostics/tracing.go [hunk 1]
```diff
@@ -30,18 +35,17 @@ const (
 
 	// span attribute keys
 	// Reference trace semantics https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/trace/semantic_conventions
-	dbTypeSpanAttributeKey      = "db.type"
-	dbInstanceSpanAttributeKey  = "db.instance"
-	dbStatementSpanAttributeKey = "db.statement"
-	dbURLSpanAttributeKey       = "db.url"
+	dbTypeSpanAttributeKey      = string(semconv.DBSystemKey)
```
Comment (youngbupark) [inline pkg/diagnostics/tracing.go:38]: Can you rename `dbTypeSpanAttributeKey` to `dbSystemSpanAttributeKey` ? please rename the other const as well..

**Review by youngbupark — CHANGES_REQUESTED** [review]

File: pkg/diagnostics/tracing.go (hunk 1 above)
Comment (tcnghia) [reply pkg/diagnostics/tracing.go]: Done

**Comment by tcnghia** [comment]
@youngbupark I want to keep this PR to contains no semantic change, only updating constants. I will follow up with another PR to make it easier to distinguish between different state stores when exporting to AppInsights.

**Review by mukundansundar — COMMENTED** [review]
With the addition of semconv from Otel, there are changes to the span names. Should this be called out separately if we are making this change?

File: go.sum [hunk 2]
```diff
@@ -1030,6 +1029,8 @@ github.com/stretchr/testify v1.3.0/go.mod h1:M5WIy9Dh21IEIfnGCwXGc5bZfKNJtfHm1UV
 github.com/stretchr/testify v1.4.0/go.mod h1:j7eGeouHqKxXV5pUuKE4zz7dFj8WfuZ+81PSLYec5m4=
 github.com/stretchr/testify v1.5.1 h1:nOGnQDM7FYENwehXlg/kFVnos3rEvtKTjRvOWSzb6H4=
 github.com/stretchr/testify v1.5.1/go.mod h1:5W2xD1RspED5o8YsWQXVCued0rvSQ+mT+I5cxcmMvtA=
+github.com/stretchr/testify v1.6.1 h1:hDPOHmpOpP40lSULcqw7IrRb/u7w6RpDC9399XyoNd0=
```
[the remark this answers is not in the document]
Comment (tcnghia) [reply go.sum:1032]: Done

**Comment by tcnghia** [comment]
> With the addition of semconv from Otel, there are changes to the span names. Should this be called out separately if we are making this change?

The span attributes we changed are called out in the issue release notes. Others are kept the same, only using constants from OpenTelemetry package.

I'm updating the PR description to call that out as well.

**Comment by mukundansundar** [comment]
@tcnghia @artursouza should this be labelled as a breaking change?

**Review by youngbupark — APPROVED** [review]
